src/main/java/alex/mojaki/s3upload/StreamTransferManager.java

```diff
@@ -502,4 +553,16 @@ public void customiseCompleteRequest(CompleteMultipartUploadRequest request) {
     public void customisePutEmptyObjectRequest(PutObjectRequest request) {
     }
 
+    private static class PartNumberComparator implements Comparator<PartETag> {
+        @Override
+        public int compare(PartETag o1, PartETag o2) {
+            int partNumber1 = o1.getPartNumber();
+            int partNumber2 = o2.getPartNumber();
+
+            if (partNumber1 == partNumber2) {
+                return 0;
+            }
+            return partNumber1 > partNumber2 ? 1 : -1;
```

Comment:
Replace body with:

```java
return Integer.compare(o1.getPartNumber(), o2.getPartNumber())
```

Reply:
This is available since java 1.7. This project uses 1.6.